source/adios2/engine/insitumpi/InSituMPISchedules.cpp

```diff
@@ -0,0 +1,318 @@
+/*
+ * Distributed under the OSI-approved Apache License, Version 2.0.  See
+ * accompanying file Copyright.txt for details.
+ *
+ * InSituMPISchedules.cpp
+ * common functions for in situ MPI Writer and Reader
+ * It requires MPI
+ *
+ *  Created on: Dec 25, 2017
+ *      Author: Norbert Podhorszki pnorbert@ornl.gov
+ */
+
+#include "InSituMPISchedules.h"
+
+#include "adios2/helper/adiosMath.h"
+#include "adios2/helper/adiosMemory.h"
+
+#include <iostream>
+
+namespace adios2
+{
+
+namespace insitumpi
+{
+
+int GetNumberOfRequests(
+    std::map<std::string, SubFileInfoMap> &variablesSubFileInfo) noexcept
```

Comment:
Since this is a Get function: const std::map<std::string, SubFileInfoMap> &variablesSubFileInfo ?